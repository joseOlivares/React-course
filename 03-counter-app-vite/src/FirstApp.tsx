


interface Props {
    title: string;
    subtitle: string;
    name: string;
}

export const FirstApp =({title, subtitle, name}: Props)  => {
  return (
    <>
        <h1 data-testid="test-title">{title}</h1>
        <p>{subtitle}</p>
        <p>{name}</p>
    </>
  )
}



