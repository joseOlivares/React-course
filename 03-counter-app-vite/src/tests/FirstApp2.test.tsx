import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe('Pruebas en <FirstApp />', () => {
    const title= 'Hola, soy Goku';
    const subtitle = 'Soy un subtitulo';
    const name= 'José Olivares';

    test('Debe de hacer match con el snapshot', () => {

        const  {container} = render(<FirstApp title={title} subtitle={subtitle} name={name}/>);
        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar el mensaje Hola, Soy Gokú', () => {   

        render(<FirstApp title={title} subtitle={subtitle} name={name}/>);
        expect(screen.getByText(title)).toBeTruthy();        

    })

    test('Debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} subtitle={subtitle} name={name}/>);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toBe(title);
    })

    test('Debe mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle} name={name}/>);
        expect(screen.getAllByText(subtitle).length).toBe(2);
    })

});