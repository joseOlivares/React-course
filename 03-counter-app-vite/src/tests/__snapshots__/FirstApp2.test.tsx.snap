// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Jos√© Olivares
  </p>
</div>
`;
